### PR TITLE
More verbose timezone name on event show page and refactoring

### LINKED ISSUE #1097: Saudi Arabia timezone issue
From one of our users

> I'm in Saudi Arabia (UTC+3).
The events listing appears to me in AST (Atlantic Standard Time). If there is an option to change this, I cannot find it. I saw in a video in CS169 that the events could appear in a different time zone than AST.

> 4 notes that might help:
* My profile: http://www.agileventures.org/users/omar-w-kh
* In the dashboard, it appears that there are no users in Saudi Arabia (I'm a user there).
* When I create an event, the site detects my time zone correctly (Asia/Riyadh).
* In the detailed view of an event. I see both AST and UTC. But not my time zone.

> It would be really helpful if I could see event listings in my time zone.

### PR DESCRIPTION
**Pull request by tansaku**
fixes #1097

### CODE REVIEW

**Comment by tansaku** [comment]
My notes from during the pairing session on this:

Problems we identified with out js:

a) loading all the javascript on all pages --> length check
b) functions in the global name space FIXED
c) ambiguous function names
d) setLocalDateAndTime function is long and not very self explanatory
e) setLocalDateAndTime gets data from DOM, manipulates date formats, and puts data back in DOM
f) selecting the user dropdown new jQuery fn is not very readable and is quite long
g) jasmine testing is at the acceptance level
h) jasmine stubbing is hand-rolled FIXED

options we discussed

*) wrap in an object (simplify public interface) EditEventFormManipulator (encapsulate knowledge of the DOM)
 
*) "single page web applications" by Mike McCofsky

----------------

Earlier notes from trying to look at all the different site priorities

1) Projects
2) Events (further polishing) --> show page for events
3) SignUp/SignIn (G+) fix G+ to support 4
4) tracking activity (investigating dashboard)
5) members with explicit karma ranking

*) repeating events with cancellations?

reward members for participation with badges, rank, profile etc.

Nick:

would like to subscribe to certain pairing events - get notifications

no social interaction mechanism on the website - must go to slack

would like 3 categories of users

1) newcomers
2) mentors (will choose users to work on projects)
3) orgs that want projects worked on

File: app/assets/javascripts/events/edit_event_form.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+var editEventForm = {
```
Comment (SuperManEver) [inline app/assets/javascripts/events/edit_event_form.js]: Great work! You've made this ugly code more understandable

File: app/assets/javascripts/events/edit_event_form.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    timezone : function() {
+        return timeZoneUtilities.detectUserTimeZone();
+    },
+    handleUserTimeZone: function () {
+        if ($("#start_time_tz").length) {
+            $('#start_time_tz').setToUserTimeZone(this.timezone());
+        }
+        if ($('#start_time').length && $('#start_date').length) {
+            this.updateDateAndTimeToUserTimeZone($('#next_date'), $('#start_date'), $('#start_time'));
```
Comment (SuperManEver) [inline app/assets/javascripts/events/edit_event_form.js]: May be there is some better abstraction. It looks like current implementation uses DOM as storage, some data stored at DOM objects (#next_data). If this function will be execute from several places each time DOM tree will be traversed multiple times in order to find this data stored in it. If tree is large there can be some performance issues. May it possible to have some model and retrieve data from there?  May be it's possible to use Local storage for that case https://developer.mozilla.org/en/docs/Web/API/Window/localStorage

File: app/assets/javascripts/events/edit_event_form.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        }
+    },
+    updateDateAndTimeToUserTimeZone : function(next_date, start_date, start_time){
+        var normalized_next_date_time = moment.utc(next_date.val() + " " + start_time.val(), "YYYY-MM-DD hh:mm a");
+        var normalized_start_date_time = moment.utc(start_date.val() + " " + start_time.val(), "YYYY-MM-DD hh:mm a");
```
Comment (SuperManEver) [inline app/assets/javascripts/events/edit_event_form.js]: may be it's possible to create some layer of indirection and elimate explicit dependency on **moment** and have some object that encapsulate all the functionality related to timezone or time on general

**Comment by tansaku** [comment]
I'm reviewing merge conflicts ...

**Comment by diraulo** [comment]
@tansaku any progress on the merge conflicts?

**Comment by tansaku** [comment]
@diraulo looking at it now :-)

**Comment by tansaku** [comment]
@diraulo - this should be ready to go - would be great to get this into develop at least as that would make it easier to work on the other related timezone bugs that have come up

**Pull request merged**